### DISCUSSION TITLE
how to let sub app use parent app's store?

### DISCUSSION THREAD

**Discussion by luqiudi** [vuejs/pinia#2589] [Help and Questions] [answered]
I'm doing a project
this project include many project
main
ext-1
ext-2
ext-3
... more than 100 extension ext type plugin vue

ext is a part of main project
but ext-1 /2/3 have to deploy in server
main project remote load ext component  from server

but ext also need load main project' store to load some state
and ext ' number is dynamic increase

so how to let  ext component get main app's store state??
ext componet is build standalone
![image](https://github.com/vuejs/pinia/assets/28947725/66e445db-cb2e-4e6e-aa71-262cb6f369bb)

**Comment by posva** [comment, marked as answer]
The important part is using **only one instance of pinia** (only one `const pinia = createPinia()` and maybe multiple `app1.use(pinia); app2.use(pinia)`. That's where the state is held. Depending on your bundling, you might want to mark pinia as external in sub dependencies

**Reply by zisvh** [reply]
Hi, any help as to how to achieve this ?

**Reply by luqiudi** [reply]
em，okay，I've solve this Requirements,
by use jsonform instead of this above pattern,
thank your very much~
😀😀😀
